gammapy/scripts/jupyter.py

```diff
@@ -0,0 +1,157 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Command line tool to perform devel management actions on jupyter notebooks."""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+from black import format_str
+import click
+import logging
+import nbformat
+import subprocess
+import time
+from ..extern.pathlib import Path
+
+log = logging.getLogger(__name__)
+
+
+@click.command(name='black')
+@click.pass_context
+def cli_jupyter_black(ctx):
+    """Format code cells with black."""
+
+    for path in build_nblist(ctx):
+        rawnb = read_notebook(path)
+
+        for cell in rawnb.cells:
+            fmt = cell['source']
+            if cell['cell_type'] == 'code':
+                try:
+                    fmt = '\n'.join(tag_magics(fmt))
+                    has_semicolon = fmt.endswith(';')
+                    fmt = format_str(src_contents=fmt,
+                                     line_length=79).rstrip()
+                    if has_semicolon:
+                        fmt += ';'
+                except Exception as ex:
+                    logging.info(ex)
+                fmt = fmt.replace("###-MAGIC COMMAND-", "")
+            cell['source'] = fmt
+
+        if rawnb:
+            nbformat.write(rawnb, str(path))
+            log.info('Jupyter notebook {} painted in black.'.format(str(path)))
+
+
+@click.command(name='stripout')
+@click.pass_context
+def cli_jupyter_stripout(ctx):
+    """Strip output cells."""
+
+    for path in build_nblist(ctx):
+        rawnb = read_notebook(path)
+
+        for cell in rawnb.cells:
+            if cell['cell_type'] == 'code':
+                cell['outputs'] = []
+
+        if rawnb:
+            nbformat.write(rawnb, str(path))
+            log.info('Jupyter notebook {} output stripped.'.format(str(path)))
+
+
+@click.command(name='execute')
+@click.pass_context
+def cli_jupyter_execute(ctx):
+    """Execute jupyter notebook."""
+
+    for path in build_nblist(ctx):
+        run_notebook(path)
+
+
+@click.command(name='test')
+@click.pass_context
+def cli_jupyter_test(ctx):
+    """Check if Jupyter notebook is broken."""
+
+    for path in build_nblist(ctx):
+        run_notebook(path)
+        rawnb = read_notebook(path)
+
+        if rawnb:
```

Comment:
I would always move the code black that does the processing into a function. In this case "test_notebook".
Then the CLI is just short, doing the loop over notebooks, read, process, write.